src/cci/cas_cci.c

```diff
@@ -712,14 +712,29 @@ cci_disconnect (int mapped_conn_id, T_CCI_ERROR * err_buf)
 
       get_last_error (con_handle, err_buf);
     }
-  else if (con_handle->broker_info[BROKER_INFO_CCI_PCONNECT] && hm_put_con_to_pool (con_handle->id) >= 0)
+  else if (con_handle->broker_info[BROKER_INFO_CCI_PCONNECT] && hm_put_con_to_pool_exceed () == 0)
     {
-      cci_end_tran_internal (con_handle, CCI_TRAN_ROLLBACK);
-      API_ELOG (con_handle, 0);
+      MUTEX_LOCK (con_handle_table_mutex);
+      if (con_handle->id < 0 || hm_put_con_to_pool (con_handle->id) < 0)
```

Comment:
If con_handle->id is less than 0, it is calling hm_delete_connection() and eventually hm_con_handle_free() function will use the wrong con_handle->id. There seems to be a lack of exception handling.

Exception handling is good, but wouldn't it be necessary to fix the part that sets the wrong value for con_handle->id , which is the root problem?

Reply:
Alas! hm_con_handle_free() is executed through hm_delete_connection -> hm_release_connection_internal. I didn't know that. Thank you!

But negative value of con_handle->id is shown when I executed experiments and It occured core dump.
So I think It must be handled for the core dump.

But I also must consider about other exception handling. Thank you.

Reply:
In this branch, this suggestion can not be applied but I will register as another issue soon.

Would you agree this choice? @airnet73 @kisoo-han